pkg/devfile/adapters/common/interface.go

```diff
@@ -8,7 +8,10 @@ import (
 type ComponentAdapter interface {
 	commandExecutor
 	Push(parameters PushParameters) error
+	Build(parameters BuildParameters) error
 	DoesComponentExist(cmpName string) (bool, error)
+	Deploy(parameters DeployParameters) error
```

Comment:
I would suggest call it `CreateDeployment` and `DeleteDeployment`

Reply:
Thanks for the feedback @girishramnani 
This may be better suited for #3478 as it was introduced there and just carried over into this PR